src/ClientInstance.cpp

```diff
@@ -384,14 +385,27 @@ PVR_ERROR CVNSIClientInstance::GetChannels(bool radio, kodi::addon::PVRChannelsR
       if (m_protocol >= 6)
       {
         std::string path = CVNSISettings::Get().GetIconPath();
-        std::string ref = vresp->extract_String();
+        std::string ref = vresp->extract_String(); // Picon ref
         if (!path.empty())
         {
           if (path[path.length() - 1] != '/')
             path += '/';
-          path += ref;
-          path += ".png";
-          tag.SetIconPath(path);
+
+          ref += ".png";
+          std::string image_path = path + ref;
+          bool exists = kodi::vfs::FileExists(image_path);
+          if (exists) // Use picon
+          {
+            tag.SetIconPath(image_path);
+          }
+          else // Use channel name
+          {
+            std::string ch = tag.GetChannelName();
+            std::transform(ch.begin(), ch.end(), ch.begin(), ::tolower);
+            ch += ".png";
+            std::string image_path = path + ch;
```

Comment:
You are redeclaring the same variable here. Instead you should simply set it here and then call `SetIconPath()` after the if statement.